build_tools/build_detect_platform

```diff
@@ -177,7 +177,7 @@ case "$TARGET_OS" in
         PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -lpthread -lrt -ldl"
         if test $ROCKSDB_USE_IO_URING; then
             # check for liburing
-            $CXX $CFLAGS -x c++ - -luring -o /dev/null 2>/dev/null  <<EOF
+            $CXX $PLATFORM_CXXFLAGS -x c++ - -luring -o /dev/null 2>/dev/null  <<EOF
```

Comment:
Will `$PLATFORM_CXXFLAGS` still include the `$CFLAGS`, if the user set it?

Reply:
Nothing has really changed as to what is written to the make_config.mk file.  AFAICT, CFLAGS end up in the PLATFORM_CXXFLAGS but the other options (CXXFLAGS, LDFLAGS, EXTRA_CXXFLAGS, EXTRA_LDFLAGS) do not get written to the file.  They are used however when doing the build/compile.